plugin/src/main/groovy/com/novoda/gradle/command/DemoModeExtension.groovy

```diff
@@ -0,0 +1,19 @@
+package com.novoda.gradle.command
+
+class DemoModeExtension {
+
+    final name
+    final extras = new HashMap<String, String>()
+
+    DemoModeExtension(name) {
+        this.name = name
+    }
+
+    def methodMissing(String name, args) {
+        extras[name] = args[0]
+    }
+
+    def propertyMissing(String name, value) {
```

Comment:
🎉 with these tricks, I am able to catch all the configs with a nice DSL.

Reply:
Can you elaborate a bit on what is happening here? Maybe we can catch-up IRL.

Reply:
It's pure groovy magic. More explanation here. 
http://groovy-lang.org/metaprogramming.html#_propertymissing